dali/imgcodec/parsers/pnm.cc

```diff
@@ -18,13 +18,60 @@ namespace dali {
 namespace imgcodec {
 
 ImageInfo PnmParser::GetInfo(ImageSource *encoded) const {
+  auto stream = encoded->Open();
+  ssize_t length = stream->Size();
+
+  // http://netpbm.sourceforge.net/doc/ppm.html
+  stream->Skip();
+  int channels = 1;
+  auto magic = stream->ReadOne<char>();
+  if ((magic == '3') || (magic == '6')) {
+    channels = 3;             // formats "p3" and "p6" are RGB color, all
+                              // other formats are bitmaps or greymaps
+  }
+
+  int state = 0;
+  enum {
+    STATE_START = 0, STATE_WIDTH = 1, STATE_HEIGHT = 2, STATE_DONE = 3
+  };
+  std::array<int, 2> dim = {0, 0};
+  auto cur = stream->ReadOne<char>();
+  do {
+    // comments can appear in the middle of tokens, and the newline at the
+    // end is part of the comment, not counted as whitespace
+    // http://netpbm.sourceforge.net/doc/pbm.html
+    if (cur == '#') {
+      do {
+        cur = stream->ReadOne<char>();
+      } while (cur != '\n');
+      cur = stream->ReadOne<char>();
+    } else if (isspace(cur)) {
+      if (++state < STATE_DONE) {
```

Comment:
How does it work if we have:
```
P1   # note the space before hash
  640  # note the spaces at the beginning of the line
```
I think that the comment in the example above should be treated as being in the middle of a whitespace block.
This should be something like (pseudocode):
```
void skip_space(stream) {
  do {
    c = stream.get();
    if (c == '#')
      SkipComment(stream)
      continue
  } while (isspace(c));
  stream.Skip(-1);  // back off by one
}

void skip_comment(stream) {
  do {
    c = stream.Read()
  } while (c != '\n')
}

int parse_int(stream) {
  int result = 0;
  do {
    c = stream.Read()
    if (isdigit(c))
      result = result * 10 + (c - '0')
    else if (c == '#')
       skip_comment(stream)
    else
      break
  }
  stream.Skip(-1)
}

skip_space(stream)
width = parse_int(stream)
skip_space(stream)
height = parse_int(stream)
```

Reply:
That is not so clear for me from the documentation, but It makes much sense.